test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -14,47 +13,85 @@ namespace Microsoft.ML.Tests.Transformers
 {
     public sealed class PcaTests : TestDataPipeBase
     {
+        private readonly ConsoleEnvironment _env;
+        private readonly string _dataSource;
+        private readonly TextSaver _saver;
+
         public PcaTests(ITestOutputHelper helper)
             : base(helper)
         {
+            _env = new ConsoleEnvironment(seed: 1, conc: 1);
+            _dataSource = GetDataPath("generated_regression_dataset.csv");
+            _saver = new TextSaver(_env, new TextSaver.Arguments { Silent = true, OutputHeader = false });
         }
 
         [Fact]
         public void PcaWorkout()
         {
-            var env = new ConsoleEnvironment(seed: 1, conc: 1);
-            string dataSource = GetDataPath("generated_regression_dataset.csv");
-            var data = TextLoader.CreateReader(env,
-                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+            var data = TextLoader.CreateReader(_env,
+                c => (label: c.LoadFloat(11), weight: c.LoadFloat(0), features: c.LoadFloat(1, 10)),
                 separator: ';', hasHeader: true)
-                .Read(dataSource);
+                .Read(new MultiFileSource(_dataSource));
 
-            var invalidData = TextLoader.CreateReader(env,
-                c => (label: c.LoadFloat(11), features: c.LoadText(0, 10)),
+            var invalidData = TextLoader.CreateReader(_env,
+                c => (label: c.LoadFloat(11), weight: c.LoadFloat(0), features: c.LoadText(1, 10)),
                 separator: ';', hasHeader: true)
-                .Read(dataSource);
+                .Read(new MultiFileSource(_dataSource));
+
+            var est = new PcaEstimator(_env, "features", "pca", rank: 4, seed: 10);
+            TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
 
-            var est = new PcaEstimator(env, "features", "pca", rank: 5, advancedSettings: s => {
-                    s.Seed = 1;
-                });
+            var est_non_default_args = new PcaEstimator(_env, "features", "pca", rank: 3, weightColumn: "weight", overSampling: 2, center: false);
```

Comment:
>est_non_default_args [](start = 16, length = 20)

please use `camelCase` for variable names, not `under_scores`

Reply:
i was talking python for some reason. fixed it :)

---
In reply to: [227066937](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227066937) [](ancestors = 227066937)